tools/hack/get-egctl.sh

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/env bash
+
+: "${BINARY_NAME:="egctl"}"
+: "${USE_SUDO:="true"}"
+: "${EGCTL_INSTALL_DIR:="/usr/local/bin"}"
```

Comment:
lets maybe install the binary in `~/.envoy-gateway/bin` ?
and then provide instructions to the user to add this to path to their `PATH` env var ?

Reply:
<img width="789" alt="image" src="https://user-images.githubusercontent.com/48784001/222039231-7978ac62-1062-4c0c-9de9-e41522ad9b00.png">
I think this is more convenient, just one command, user can install eg quickly. Don`t need them to add it their `PATH`.

Reply:
doesnt this require root permission ? 
how about making it explicit ? using a `-b` flag , which by default is set to
`~/.envoy-gateway/bin` and can be explicitly set to `/usr/local/bin`
example on https://func-e.io

Reply:
<img width="676" alt="image" src="https://user-images.githubusercontent.com/48784001/223916180-1c61373a-5171-4c9e-abb9-aee22d055ac3.png">
This can automatically ask end users to type pwd if has not root permission.

I have tried func-e, it will not ask permission from users, and it will fail.
<img width="666" alt="image" src="https://user-images.githubusercontent.com/48784001/223916357-9cec99fe-3cc1-4163-a6cc-d3fa8093c7d1.png">